doc/authoring_command_modules/authoring_commands.md

```diff
@@ -148,8 +148,7 @@ Now the user may identify the target IP config by specifying either the resource
 
 A couple things to note:
 - Currently, `--ids` is not exposed for any command that is called 'create', even if it is configured properly.
-- The supported resource-name values for `id_part` are: `name` and `child_name_{level}`; for example, `child_name_2` is the name of the child two levels under the resource correspond to `name`.
-- Values also available are `child_type_{level}`, `child_namespace_{level}`, and `child_parent_{level}`. Note that these values may not exist for the particular id being parsed.
+- `id_part` accepts `name` for the name of the root resource. For child resources, `child_name_X`, `child_type_X`, and `child_namespace_X` are allowed where X is an integer indicating the child position relative to the root resources (larger integers are further removed). Type and namespace are optional and, if not supplied, essentially inherit from the nearest upstream relative.
```

Comment:
Type is NOT optional. Every resource/child name will have an associated type. 
Namespace is optional, but does not inherit from the its parent. This allows us to differentiate between the following ids:
`/subscriptions/sub/resourcegroups/group/providers/namespace/rootType/rootName/childType/childName`
and
`/subscriptions/sub/resourcegroups/group/providers/namespace/rootType/rootName/providers/childNamespace/childType/childName`
If the namespace was inherited, and the id was reformed, it becomes unclear whether the inherited namespace would be included in the resulting id without checking the child's parent.

Reply:
Good point!